### PR TITLE
Boost thread header for txdb.

### PR DESCRIPTION
**Pull request by chrisandrewca**
Hello! Adding this boost header fixes the WSL windows build. Probably not the right place for it but a good conversation starter?

### CODE REVIEW

**Comment by domob1812** [comment]
Thanks!  What is the error message you get without this?  I just want to better understand the underlying issue.

**Comment by chrisandrewca** [comment]
The error I came across is interruption_point is not a function declared in boost::this_thread.

https://github.com/namecoin/namecoin-core/blob/e25eaf14ebae99f99d6c9ad6edf9e56696144d8d/src/txdb.cpp#L365

**Comment by domob1812** [comment]
Ah I see.  Then I think the proper fix would be to get rid of the `boost::this_thread::interruption_point` instead.  This is a legacy thing, that was changed in other parts of the code in upstream Bitcoin already.  I think what we should do is this:

- Replace that boost call by a passed-in closure, similar to how e.g. [`GetUTXOStats`](https://github.com/namecoin/namecoin-core/blob/2f2bfb323e7e19512697039753c0675dacbb6dec/src/node/coinstats.cpp#L120) does it.
- In the call from `names/main.cpp`, just pass an empty function.
- In the call from the RPC code, [pass in `node.rpc_interruption_point`](https://github.com/namecoin/namecoin-core/blob/2f2bfb323e7e19512697039753c0675dacbb6dec/src/rpc/blockchain.cpp#L1074).

Do you feel like doing that, or would you rather have me do it?  It should not be much work, but a bit more than this PR.  (But then I think it is the proper fix, cleaning up some legacy artefact.)

**Comment by chrisandrewca** [comment]
I'm happy to take a look at that work. How do you track divergence from upstream Bitcoin? Would a merge of Bitcoin proper be something to consider?

**Comment by domob1812** [comment]
We merge upstream changes in Bitcoin weekly, and try to keep as close to it as possible.  In this case, the code in question is specific to Namecoin.  It was written years ago based on how upstream Bitcoin iterated through the UTXO set at the time, but then the upstream code got changed and the Namecoin code stayed.  That's why I think the proper fix now is to change the Namecoin code to be back in line with how upstream Bitcoin handles interruptions.

**Comment by chrisandrewca** [comment]
Thanks for the downlo. Alright I gave it a go, check er out.

File: src/names/main.cpp [hunk 1]
```diff
@@ -484,7 +484,7 @@ CheckNameDB (ChainstateManager& chainman, bool disconnect)
 
   auto& coinsTip = chainman.ActiveChainstate ().CoinsTip ();
   coinsTip.Flush ();
-  const bool ok = coinsTip.ValidateNameDB (chainman);
+  const bool ok = coinsTip.ValidateNameDB (chainman, [](){});
```
Comment (domob1812) [inline src/names/main.cpp:487]: Tiny nit:  Please format the lambda as `[] () {}` (with spaces).

File: src/rpc/names.cpp [hunk 2]
```diff
@@ -1054,10 +1055,12 @@ name_checkdb (const JSONRPCRequest& request)
   ).Check (request);
 
   ChainstateManager& chainman = EnsureChainman (request.context);
+  NodeContext& node = EnsureNodeContext(request.context);
```
Comment (domob1812) [inline src/rpc/names.cpp:1058]: Tiny nit:  Please add a space before the function-call parenthesis to match the style in this file.

**Review by domob1812 — APPROVED** [review]
ACK 9e2b6aa2c5c581c615dc35119aa18210f8120c5b, except for the two tiny formatting nits.  Also please clean up the commit history, i.e. drop the previous commit so it is just the proper fix.  Then I'm happy to merge this, thanks a lot!

**Comment by chrisandrewca** [comment]
Can do!

**Comment by chrisandrewca** [comment]
Hey @domob1812, this one is cleaned up.

**Comment by domob1812** [comment]
Thanks!  ACK 080a54d0f9791d3c4f555b552aa8afc58c4637c1.

**Pull request merged**